apis/actions.github.com/v1alpha1/autoscalingrunnerset_types.go

```diff
@@ -80,7 +82,98 @@ type AutoscalingRunnerSetSpec struct {
 
 type GitHubServerTLSConfig struct {
 	// Required
-	RootCAsConfigMapRef string `json:"certConfigMapRef,omitempty"`
+	CertificateFrom *TLSCertificateSource `json:"certificateFrom,omitempty"`
+
+	// +optional
+	// +kubebuilder:default:=/usr/local/share/ca-certificates/
+	RunnerMountPath string `json:"runnerMountPath,omitempty"`
+}
+
+func (c *GitHubServerTLSConfig) ToCertPool(keyFetcher func(name, key string) ([]byte, error)) (*x509.CertPool, error) {
+	if c == nil {
```

Comment:
Should we guard against the nil receiver? I feel like it should be callers responsibility to check if GitHubServerTLSConfig is set before calling this method